### PR TITLE
feat: Phase 1 — migrate read queries to reservation tables (PG)

### PR DESCRIPTION
**Pull request by porcellus**
## Summary

- Migrate all read queries off `all_auth_recipe_users` and per-recipe `*_user_to_tenant` tables to reservation tables (`recipe_user_tenants`, `recipe_user_account_infos`, `primary_user_tenants`, `app_id_to_user_id`)
- Migrate `getUsers()` dashboard search to unified reservation table queries with proper search tag combinations
- Migrate `listUsersByAccountInfo`, `getPrimaryUserInfo`, `getTenantIds`, session queries, and `doesUserIdExist`
- Fix Primery typo and optimize `lockUsers` to single FOR UPDATE query
- Fix Phase 1 bugs: `phone_number` column name, dual email+phone passwordless reservations, `linkAccounts` time_joined ordering, search tag edge cases, CHAR(36) padding in lockUsers
- Add comprehensive reservation table integrity checker (I1-I6) and 10 new tests

**Phase 1 of the DEPRECATE plan** — moves all read paths to the new reservation tables. After this, the old tables are only used for writes (Phase 2 will address those).

**Companion PRs:**
- supertokens/supertokens-plugin-interface#197
- supertokens/supertokens-core#1249

## Test plan
- [ ] Full test suite (running)
- [ ] `ReservationTableIntegrityTest` — 10 tests verifying all 6 invariants across email, phone, third-party
- [ ] `ExceptionParsingTest` — integrity checks added to signup methods
- [ ] 6 race test classes upgraded from email-only to full reservation consistency checks
- [ ] Lint passes (`spotlessJavaCheck` clean)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/main/java/io/supertokens/storage/postgresql/queries/SessionQueries.java [hunk 1]
```diff
@@ -90,11 +91,13 @@ static String getQueryToCreateAccessTokenSigningKeysTable(Start start) {
         // @formatter:on
     }
 
+    // TODO: Add IF NOT EXISTS to prevent crash on dirty DB state from prior test failures
```
Comment (tamassoltesz) [inline src/main/java/io/supertokens/storage/postgresql/queries/SessionQueries.java:94]: TODO

File: src/main/java/io/supertokens/storage/postgresql/queries/SessionQueries.java [hunk 2]
```diff
@@ -90,11 +91,13 @@ static String getQueryToCreateAccessTokenSigningKeysTable(Start start) {
...
     public static String getQueryToCreateAppIdIndexForAccessTokenSigningKeysTable(Start start) {
         return "CREATE INDEX access_token_signing_keys_app_id_index ON "
                 + Config.getConfig(start).getAccessTokenSigningKeysTable() + "(app_id);";
     }
 
+    // TODO: Add IF NOT EXISTS to prevent crash on dirty DB state from prior test failures
```
Comment (tamassoltesz) [inline src/main/java/io/supertokens/storage/postgresql/queries/SessionQueries.java:100]: TODO

File: src/main/java/io/supertokens/storage/postgresql/queries/UserMetadataQueries.java [hunk 3]
```diff
@@ -57,6 +57,7 @@ public static String getQueryToCreateUserMetadataTable(Start start) {
         // @formatter:on
     }
 
+    // TODO: Add IF NOT EXISTS to prevent crash on dirty DB state from prior test failures
```
Comment (tamassoltesz) [inline src/main/java/io/supertokens/storage/postgresql/queries/UserMetadataQueries.java:60]: TODO

File: src/main/java/io/supertokens/storage/postgresql/queries/UserRolesQueries.java [hunk 4]
```diff
@@ -52,6 +52,7 @@ public static String getQueryToCreateRolesTable(Start start) {
         // @formatter:on
     }
 
+    // TODO: Add IF NOT EXISTS to prevent crash on dirty DB state from prior test failures
```
Comment (tamassoltesz) [inline src/main/java/io/supertokens/storage/postgresql/queries/UserRolesQueries.java:55]: TODO

File: src/main/java/io/supertokens/storage/postgresql/queries/UserRolesQueries.java [hunk 5]
```diff
@@ -73,11 +74,13 @@ public static String getQueryToCreateRolePermissionsTable(Start start) {
...
```
Comment (tamassoltesz) [inline src/main/java/io/supertokens/storage/postgresql/queries/UserRolesQueries.java:77]: TODO

File: src/main/java/io/supertokens/storage/postgresql/queries/UserRolesQueries.java [hunk 6]
```diff
@@ -73,11 +74,13 @@ public static String getQueryToCreateRolePermissionsTable(Start start) {
...
     public static String getQueryToCreateRoleIndexForRolePermissionsTable(Start start) {
         return "CREATE INDEX role_permissions_role_index ON " + getConfig(start).getUserRolesPermissionsTable()
                 + "(app_id, role);";
     }
 
+    // TODO: Add IF NOT EXISTS to prevent crash on dirty DB state from prior test failures
```
Comment (tamassoltesz) [inline src/main/java/io/supertokens/storage/postgresql/queries/UserRolesQueries.java:83]: TODO

File: src/test/java/io/supertokens/storage/postgresql/test/DatabaseTestHelper.java [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+            System.err.println("[DatabaseTestHelper] Failed to create database " + dbName + ": " + e.getMessage());
+            throw new RuntimeException("Failed to create test database: " + dbName, e);
+        }
+    }
+
+    /**
+     * Clear the current test database reference.
+     * The database persists and is reused for subsequent tests in this worker.
+     */
+    public static void dropCurrentTestDatabase() {
```
Comment (tamassoltesz) [inline src/test/java/io/supertokens/storage/postgresql/test/DatabaseTestHelper.java:130]: I think a `remove..` or something similar name would be better, considering the first comment in the first line of the method directly contradicts the name :D

**Review by tamassoltesz — CHANGES_REQUESTED** [review]

**Comment by porcellus** [comment]
Superseded by #322, which consolidates feat/isolation-reservation-table, feat/isolation-reservation-table-locking, feat/isolation-reservation-table-migration, feat/deprecate-phase-2-migration-mode, and refactor/performance_improvements into a single PR targeting master. Closing in favor of that PR — please re-route review there.
